functions/webhooks/taskrouterCallback.protected.ts

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
+import '@twilio-labs/serverless-runtime-types';
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import { responseWithCors, bindResolve, success, error500 } from '@tech-matters/serverless-helpers';
+
+// eslint-disable-next-line prettier/prettier
+import type { AddCustomerExternalId } from '../helpers/addCustomerExternalId.private';
+// eslint-disable-next-line prettier/prettier
+import type { PostSurveyJanitor } from '../helpers/postSurveyJanitor.private';
+
+export type Body = {
+  EventType: string;
+  TaskSid?: string;
+  TaskAttributes?: string;
+};
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  CHAT_SERVICE_SID: string;
+  FLEX_PROXY_SERVICE_SID: string;
+};
+
+const TASK_CREATED_EVENT = 'task.created';
+const TASK_CANCELED_EVENT = 'task.canceled';
+const TASK_COMPLETED_EVENT = 'task.completed';
+
+const wait = (ms: number): Promise<void> => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+};
+
+export const handler = async (
```

Comment:
This file could use some comments to give a quick summary of what we're doing with each event type.  This is only going to get bigger, and keeping a comment block of how we're routing events will be really important to understanding what's happening.